nav2_collision_monitor/include/nav2_collision_monitor/polygon_velocity.hpp

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) 2022 Dexory
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef NAV2_COLLISION_MONITOR__POLYGON_VELOCITY_HPP_
+#define NAV2_COLLISION_MONITOR__POLYGON_VELOCITY_HPP_
+
+#include <string>
+#include <vector>
+
+#include "rclcpp/rclcpp.hpp"
+
+#include "nav2_collision_monitor/types.hpp"
+
+namespace nav2_collision_monitor
+{
+
+/**
+ * @brief Polygon velocity class.
+ * This class contains all the points of the polygon and
+ * the expected condition of the velocity based polygon.
+ */
+class PolygonVelocity
```

Comment:
VelocityPolygon

I think the main thing here is that this is a velocity profile to apply a polygon, not a polygon to apply a velocity profile. The conditional input is the velocity which we apply the attribute of the polygon.